appinventor/components/src/com/google/appinventor/components/runtime/Chart.java

```diff
@@ -0,0 +1,457 @@
+    }
+
+    /**
+     * Specifies the text displayed by the description label.
+     *
+     * @param text  description
+     */
+    @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_STRING,
+            defaultValue = "")
+    @SimpleProperty
+    public void Description(String text) {
+        this.description = text;
+        chartView.setDescription(description);
+    }
+
+    /**
+     * Returns the chart's background color as an alpha-red-green-blue
+     * integer.
+     *
+     * @return  background RGB color with alpha
+     */
+    @SimpleProperty(
+            category = PropertyCategory.APPEARANCE)
+    public int BackgroundColor() {
+        return backgroundColor;
+    }
+
+    /**
+     * Specifies the chart's background color as an alpha-red-green-blue
+     * integer.
+     *
+     * @param argb  background RGB color with alpha
+     */
+    @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_COLOR,
+            defaultValue = Component.DEFAULT_VALUE_COLOR_NONE)
+    @SimpleProperty
+    public void BackgroundColor(int argb) {
+        backgroundColor = argb;
+        chartView.setBackgroundColor(argb);
+    }
+
+    /**
+     * Sets the Pie Radius of the Chart. If the current type is
+     * not the Pie Chart, the value is simply stored, but not
+     * processed.
+     *
+     * The value is hidden in the blocks due to it being applicable
+     * to a single Chart only. TODO: Might be better to change this in the future
+     *
+     * TODO: Make this an enum selection in the future? (Donut, Full Pie, Small Donut, etc.)
+     * @param percent  Percentage of the Pie Chart radius to fill
+     */
+    @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_CHART_PIE_RADIUS,
+                defaultValue = "100")
+    @SimpleProperty(userVisible = false)
+    public void PieRadius(int percent) {
+        this.pieRadius = percent;
+
+        // Only set the value if the Chart View is a
+        // Pie Chart View; Otherwise take no action.
+        if (chartView instanceof PieChartView) {
+            ((PieChartView)chartView).setPieRadius(percent);
+        }
+    }
+
+    /**
+     * Changes the visibility of the Chart's Legend.
+     * @param enabled  indicates whether the Chart should be enabled.
+     */
+    @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_BOOLEAN,
+                defaultValue = "True")
+    @SimpleProperty
+    public void LegendEnabled(boolean enabled) {
+        this.legendEnabled = enabled;
+        chartView.setLegendEnabled(enabled);
+    }
+
+    /**
+     * Changes the visibility of the Chart's grid, if the
+     * Chart View is a Chart with an Axis.
+     * @param enabled  indicates whether the Chart's grid should be enabled.
+     */
+    @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_BOOLEAN,
+        defaultValue = "True")
+    @SimpleProperty
+    public void GridEnabled(boolean enabled) {
```

Comment:
@ewpatton Done, however, I'm a bit worried because some Chart types don't have a grid (Pie Chart), and there is no way to hide blocks directly as of now AFAIK